EIPS/eip-7702.md

```diff
@@ -40,38 +41,67 @@ We introduce a new [EIP-2718](./eip-2718.md) transaction, "set code transaction"
 
 #### Behavior
 
-At the start of executing the transaction, for each `[chain_id, address, [nonce], y_parity, r, s]` tuple:
+At the start of executing the transaction, for each `[chain_id, address, y_parity, r, s]` tuple:
 
-1. `authority = ecrecover(keccak(MAGIC || rlp([chain_id, [nonce], address])), y_parity, r, s]`
+1. `authority = ecrecover(keccak(MAGIC || rlp([chain_id, address])), y_parity, r, s)`
 2. Verify the chain id is either 0 or the chain's current ID.
 3. Verify that the code of `authority` is empty.
-4. If nonce list item is length one, verify the nonce of `authority` is equal to `nonce`.
+4. Verify that the storage slot `keccak(authority || address)` of account `AUTH_REVOCATION_ADDRESS` is empty. (`authority` and `address` should be padded to 32 bytes each.)
 5. Set the code of `authority` to code associated with `address`.
 6. Add the `authority` account to `accessed_addresses` (as defined in [EIP-2929](./eip-2929.md).)
 
-If any of the above steps fail, immediately stop processing that tuple and continue to the next tuple in the list. It will In the case of multiple tuples for the same authority, set the code specified by the address in the first occurrence.
+If any of the above steps fail, immediately stop processing that tuple and continue to the next tuple in the list. In the case of multiple tuples for the same authority, set the code specified by the address in the first occurrence.
 
 At the end of the transaction, set the code of each `authority` back to empty.
 
 Note that the signer of an authorization tuple may be different than `tx.origin` of the transaction.
 
 #### Gas Costs
 
-The intrinsic cost of the new transaction is inherited from [EIP-2930](./eip-2930.md), specifically `21000 + 16 * non-zero calldata bytes + 4 * zero calldata bytes + 1900 * access list storage key count + 2400 * access list address count`. Additionally, we add a cost of `PER_CONTRACT_CODE_BASE_COST * authorization list length`.
+The intrinsic cost of the new transaction is inherited from [EIP-2930](./eip-2930.md), specifically `21000 + 16 * non-zero calldata bytes + 4 * zero calldata bytes + 1900 * access list storage key count + 2400 * access list address count`. Additionally, we add a cost of `PER_AUTH_BASE_COST * authorization list length`.
 
 The transaction sender will pay for all authorization tuples, regardless of validity or duplication.
 
+### Revocation Contract
+
+In order to revoke a previously signed authorization, the signer can invoke the revocation contract at `AUTH_REVOCATION_ADDRESS` with the address to be revoked in calldata, padded to 32 bytes. Revocation is permanent. Previously valid signatures for the address will be ignored in all subsequent set-code transactions.
+
+The assembly code of the revocation contract is:
+
+```
+// write caller to [0, 32)
+caller
+push0
+mstore
+
+// copy address argument to [32, 64)
+push0
+calldataload
+push1 0x20
+mstore
+
+push1 0x01
```

Comment:
In the current version of the proposal the revocation can be made by account code (so not necessarily made by the EOA directly). Since we are working under a threat model where this code can become malicious, I thought it best to make revocation permanent so that a potentially untrusted template is not able to undo past revocations.

It seems easier to reason about the security properties in this way.